src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponentSpec.scala

```diff
@@ -0,0 +1,437 @@
+
+          //verify it was actually copied into the workspace
+          assert(runAndWait(entityQuery.list(context1, "SampleSet")).toList.contains(x1))
+          assert(runAndWait(entityQuery.list(context1, "SampleSet")).toList.contains(x2))
+        }
+      }
+
+  }
+
+  it should "copy entities without a conflict with a cycle" in withDefaultTestDatabase {
+
+    runAndWait(workspaceQuery.save(workspace2))
+    withWorkspaceContext(testData.workspace) { context1 =>
+      withWorkspaceContext(workspace2) { context2 =>
+        val a = Entity("a", "test", Map.empty)
+        val a6 = Entity("a6", "test", Map("next" -> AttributeEntityReference("test", "a1")))
+        val a5 = Entity("a5", "test", Map("next" -> AttributeEntityReference("test", "a6")))
+        val a4 = Entity("a4", "test", Map("next" -> AttributeEntityReference("test", "a5")))
+        val a3 = Entity("a3", "test", Map("next" -> AttributeEntityReference("test", "a4"), "side" -> AttributeEntityReference("test", "a")))
+        val a2 = Entity("a2", "test", Map("next" -> AttributeEntityReference("test", "a3")))
+        val a1 = Entity("a1", "test", Map("next" -> AttributeEntityReference("test", "a2")))
+        runAndWait(entityQuery.save(context2, a))
+        runAndWait(entityQuery.save(context2, a6.copy(attributes = Map.empty)))
+        runAndWait(entityQuery.save(context2, a5))
+        runAndWait(entityQuery.save(context2, a4))
+        runAndWait(entityQuery.save(context2, a3))
+        runAndWait(entityQuery.save(context2, a2))
+        runAndWait(entityQuery.save(context2, a1))
+
+        // the cycle
+        runAndWait(entityQuery.save(context2, Entity("a6", "test", Map("next" -> AttributeEntityReference("test", "a1")))))
+
+        // note: we're copying FROM workspace2 INTO workspace
+        assertResult(Seq.empty) {
+          runAndWait(entityQuery.getCopyConflicts(context1, Seq(a1, a2, a3, a4, a5, a6, a)))
+        }
+
+        assertResult(Seq.empty) {
+          runAndWait(entityQuery.copyEntities(context2, context1, "test", Seq("a1")))
+        }
+
+        //verify it was actually copied into the workspace
+        assertResult(Set(a1, a2, a3, a4, a5, a6, a)) {
+          runAndWait(entityQuery.list(context1, "test")).toSet
+        }
+      }
+    }
+
+  }
+
+  val y1 = Entity("y1", "SampleSet", Map("ref" -> AttributeEntityReference("SampleSet", "y2")))
+  var y2 = Entity("y2", "SampleSet", Map.empty)
+
+
+  it should "save an entity without attribute references and then save that entity again with attribute references" in withDefaultTestDatabase { 
+
+      withWorkspaceContext(testData.workspace) { context =>
+        runAndWait(entityQuery.save(context, y2))
+        runAndWait(entityQuery.save(context, y1))
+        y2 = Entity("y2", "SampleSet", Map("ref" -> AttributeEntityReference("SampleSet", "y1")))
+        runAndWait(entityQuery.save(context, y2))
+
+        assert(runAndWait(entityQuery.list(context, "SampleSet")).toList.contains(y1))
```

Comment:
I think I might remove this test as it is recapitulated in copy entities without a conflict with a cycle. But specifically this test was written when fixing a bug that would delete y1 while saving y2... IIRC.